### PR TITLE
feat: add extension methods we use on the ADM project

### PR DESCRIPTION
**Pull request by OwenLacey28**
| Action | Done? |
| --- | --- |
| Code compiles and unit tests all pass locally | ✔ |
| Debug/console log code removed | ✔ |
| Commented out code removed |  ✔ |
| Unit tests added/updated | ✔ - ish |
| README updated | ❌ |
| Considered: <br /> - Performance <br /> - Security <br /> - Logging | ❌ |
| Licenses of any new/upgraded dependencies checked, with no copyleft dependencies introduced | ❌ |

image

### CODE REVIEW

File: src/Audacia.Core/Extensions/ExpressionEnumerableExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    {
+        /// <summary>
+        /// Reduce multiple expressions with AND logic into a single expression.
+        /// </summary>
+        /// <param name="expressions">The list of expressions to combine.</param>
+        /// <typeparam name="T">The type of the source of the predicate.</typeparam>
+        /// <returns>A combined expression representing the OR logic of all input expressions.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="expressions"/> is null.</exception>
+        public static Expression<Func<T, bool>> Any<T>(
+            this List<Expression<Func<T, bool>>> expressions)
```
Comment (Audacia-RhysSmith) [inline src/Audacia.Core/Extensions/ExpressionEnumerableExtensions.cs]: 🔧 Does this want to be modify to` IEnumerable<T>` instead of `List<T>`

**Review by Audacia-RhysSmith — CHANGES_REQUESTED** [review]

File: src/Audacia.Core/Extensions/ExpressionEnumerableExtensions.cs (hunk 1 above)
Comment (OwenLacey28) [reply src/Audacia.Core/Extensions/ExpressionEnumerableExtensions.cs]: good point - done. 

I've also simplified it a lot with `Aggregate`

Comment (Audacia-RhysSmith) [reply src/Audacia.Core/Extensions/ExpressionEnumerableExtensions.cs]: Nice 👍

**Review by Audacia-RhysSmith — APPROVED** [review]

**Review by Audacia-RhysSmith — CHANGES_REQUESTED** [review]
Un approving until version is incremented

**Review by Audacia-RhysSmith — APPROVED** [review]

**Pull request merged**